### ISSUE TITLE
🐛 BUG: fetch silently restarts server if it hangs

### ISSUE THREAD

**Issue by janwilmake** [cloudflare/workers-sdk#7289]
### Which Cloudflare product(s) does this pertain to?

Wrangler

### What version(s) of the tool(s) are you using?

3.87.0

### What version of Node are you using?

20.17.0

### What operating system and version are you using?

macOS Sonoma Version 14.6.1 (23G93)

### Observed behavior
If you do a fetch request but it is blocked by a firewall (I'm using little snitch) it silently fails and restarts the server, somehow ending up in the same place again.

It restarts about 5x per second.

In my terminal, I see this:

```
Reloading local server...
Reloading local server...
Reloading local server...
Reloading local server...
Reloading local server...
Reloading local server...
// infinitely
```

Also, this is what the curl -v shows:
```
➜  ~ curl -v 'http://localhost:3000/repos/facebook/react' \
-H 'Authorization: Bearer kfjsdlkfjdsklfjdslkjflkdsjflkjdlkjfskljdflkjfskljdfkljskljfslkjfdskl' \
-H 'Accept: application/vnd.github.v3+json'
* Host localhost:3000 was resolved.
* IPv6: ::1
* IPv4: 127.0.0.1
*   Trying [::1]:3000...
* Connected to localhost (::1) port 3000
> GET /repos/facebook/react HTTP/1.1
> Host: localhost:3000
> User-Agent: curl/8.7.1
> Authorization: Bearer kfjsdlkfjdsklfjdslkjflkdsjflkjdlkjfskljdflkjfskljdfkljskljfslkjfdskl
> Accept: application/vnd.github.v3+json
> 
* Request completely sent off
```

It hangs at "Request completely sent off" without any further details. The server keeps restarting.

Even after completely turning off `little snitch` and restarting the terminal, this problem still happens. I also tried restarting my computer without luck. 

It seems to be unrelated to the firewall. Even with little snitch turned off, it still happens.

It's something introduced in the past 2 months. When I went back to `npx wrangler@3.78.10 dev --port 3000` I didn't have this problem, and it works smoothly, also with little snitch. 

To get started, I think it would be good to take a look what has changed around `fetch` in the local environment in the last 2 months. In production, it works fine.

### Expected behavior
The expected behavior is that the `fetch` request will not silently fail and keep restarting the local server in an infinite loop.

**Comment by penalosa** [comment]
Thanks for reporting this! Would you be able to provide a reproduction repo that demonstrates the issue? In particular, it sounds like you're seeing this behaviour _without_ a firewall as well as with one—could I confirm whether it's related to a firewall?

**Comment by CodeFromAnywhere** [comment]
It doesn't seem to be related to the firewall. I thought that at first, but the issue kept happening without firewall. I'm now on `wrangler@3.78.10` and there it works fine, also with `little snitch` enabled.

I logged in my worker and the last log happened before a `fetch` call , then the worker restarted.

I tried making a minimal example but when I started removing other things (such as r2) it didn't happen anymore that it restarted at the fetch. So I can't share any reproducible example. If needed however, I can maybe share you my worker code in private.

**Comment by CarmenPopoviciu** [comment]
hi @CodeFromAnywhere, so just to confirm... you were experiencing the issue on `wrangler@3.78.0` but no longer experience it on `wrangler@3.78.10`? Is that correct?

**Comment by lrapoport-cf** [comment]
hi @janwilmake :) we haven't heard from you in a while and it looks like you are no longer experiencing this issue, so we're going to close this for now. if this is still a problem, please feel free to open a new issue referencing this one and include an updated summary and we will revisit and investigate further. thanks!
